src/mercury_engine_data_structures/formats/bcskla.py

```diff
@@ -0,0 +1,167 @@
+
+import construct
+from construct.core import (
+    Array,
+    Computed,
+    Const,
+    Construct,
+    Container,
+    FocusedSeq,
+    If,
+    IfThenElse,
+    Int8ul,
+    Int16ul,
+    Int32ul,
+    ListContainer,
+    Pointer,
+    Rebuild,
+    Struct,
+    Tell,
+    this,
+)
+
+from mercury_engine_data_structures.common_types import Float, VersionAdapter
+from mercury_engine_data_structures.construct_extensions.alignment import AlignTo
+from mercury_engine_data_structures.formats.base_resource import BaseResource
+from mercury_engine_data_structures.formats.property_enum import PropertyEnumDoubleUnsafe
+from mercury_engine_data_structures.game_check import Game
+
+
+def _rebuild_flags(ctx):
+    flags = 0
+    for i, val in enumerate(ctx.data):
+        if isinstance(val, ListContainer):
+            flags += 2 ** i
+    return flags
+
+def _update_next_kfv(ctx):
+    ctx._._._._next_kfv_offset = ctx._new_eof
+
+class DreadKFVAdapter(construct.Adapter):
+    SUBCON = Struct(
+        timing_type = Int16ul,
```

Comment:
this should probably use an `EnumAdapter` even if it's obscured underneath an adapter